### PR TITLE
fix '\n' encoding

### PR DESCRIPTION
**Pull request by twang3**

### CODE REVIEW

File: lib/urlencode.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ function encode(str, charset) {
   var buf = iconv.encode(str, charset);
   var encodeStr = '';
   for (var i = 0; i < buf.length; i++) {
-    encodeStr += '%' + buf[i].toString('16');
+    encodeStr += '%' + buf.toString('hex', i, i + 1);
```
Comment (fengmk2) [inline lib/urlencode.js]: What's different between these two ways?

Comment (twang3) [inline lib/urlencode.js]: for '\n', "buf[i].toString('16')" get "a", but "buf.toString('hex', i, i + 1)" get "0a".

**Comment by fengmk2** [comment]
@twang3 Thanks!

**Comment by fengmk2** [comment]
1.0.1

**Comment by twang3** [comment]
my pleasure!

**Pull request merged**